R/tidy-upkeep.R

```diff
@@ -30,16 +68,67 @@ use_tidy_upkeep_issue <- function(year = NULL) {
   view_url(issue$html_url)
 }
 
-upkeep_checklist <- function(year = NULL,
-                             posit_pkg = is_posit_pkg(),
-                             posit_person_ok = is_posit_person_canonical()) {
-  year <- year %||% 2000
+upkeep_checklist <- function(year = NULL) {
+
+  bullets <- c(
+    year,
+    "",
+    todo("`usethis::use_readme_rmd()`", !file_exists("README.Rmd")),
+    todo("`usethis::use_roxygen_md()`", !is_true(uses_roxygen_md())),
+    todo("`usethis::use_github_links()`", !has_github_links()),
+    todo("`usethis::use_pkgdown_github_pages()`", !uses_pkgdown()),
+    todo("`urlchecker::url_check()`"),
+    todo(
+      "
+        `usethis::use_package_doc()`
+        Consider letting usethis manage your `@importFrom` directives here.
+        `usethis::use_import_from()` is handy for this.",
+      !has_package_doc()
+    ),
+    todo(
+      "
+        `usethis::use_testthat(3)` and upgrade to 3e, \\
+        [testthat 3e vignette](https://testthat.r-lib.org/articles/third-edition.html)",
+      !uses_testthat_edition(3)
```

Comment:
I haven't done any analysis of this but ... what's going to happen to someone who just has no tests at all?

Reply:
It will return a `FALSE` and thus the bullet will appear - I did think about having two separate bullets - one to tell people to set up tests if they didn't have any at all, and one to upgrade testthat if it is < 3, but I thought this one would capture both sufficiently.

Reply:
It probably wouldn't be too hard to distinguish "uses testthat (but not 3e)" from "doesn't use testthat" and customize the bullet accordingly.

These are pretty different audiences, I think, so maybe worth giving them their own bullet point, hyperlink, etc.

Reply:
Ok, great, I'll flesh that out.

Reply:
See fcc9ffd and 6992786